src/components/typography/Typography.tsx

```diff
@@ -0,0 +1,13 @@
+import { Typography as MuiTypography } from "@mui/material";
+import { typographySx } from "./typography.styles";
+
+import type { TypographyProps } from "./types";
+
+export default function Typography(props: TypographyProps) {
+  return (
+    <MuiTypography
+      sx={props.fontType && typographySx[props.fontType]}
```

Comment:
How about refactor the logic to `helpers.ts` file
May be we need to handle logic like this

```typescript
function getCustomVariantProp(customVariant) {
  switch (customVariant) {
    case 'regular-displayLG'
      return { component: 'p', sx: { customVariant[customVariant] } };
    // ...
    default:
      throw new Error('unknown custom variant type');
  }
}
```

Reply:
Handle logic if user need to add their custom component